data_processing/eia.py

```diff
@@ -0,0 +1,1526 @@
+                                            'Reported Prime Mover',
+                                            'Reported Fuel Type Code',
+                                            netGen])
+
+    universities = label_generation['NAICS Code'] == 611
+
+    label_generation = label_generation.loc[universities]
+
+    labels = label_generation['Reported Fuel Type Code'].unique().tolist()
+
+    base_dict = dict()
+
+    for fuel in labels:
+
+        base_dict.update({fuel: 0})
+
+    data = generation(inp)
+
+    for university in data:
+
+        del data[university]['Total Generation']
+
+        for fuel_type in data[university]:
+
+            if sort == 'installed':
+
+                base_dict[fuel_type] += 1
+
+            elif sort == 'active':
+
+                summ = 0
+
+                for mover in data[university][fuel_type]:
+
+                    if data[university][fuel_type][mover] != 0:
+
+                        summ += 1
+
+                if summ != 0:
+
+                    base_dict[fuel_type] += 1
+
+    return base_dict
+
+
+def capacity_factors():
+    """
+    This function generates a hard-coded dictionary
+    containing the capacity factors for different fuels
+    and movers. The values are defaulted to 1.0 if no
+    capacity factor information can be found.
+
+    Returns:
+    --------
+    cf: dict
+        A dictionary containing the capacity factors for
+        different fuels and movers.
+
+    """
+
+    cf = {
+        'DFO': {
+            'IC': 0.019,
+            'GT': 0.013,
+            'ST': 0.142,
+            'CA': 0.142,
+            'CT': 0.019
+        },
+        'NG': {
+            'IC': 0.13,
+            'GT': 0.119,
+            'CA': 0.55,
+            'CT': 0.55,
+            'ST': 0.126,
+            'CS': 0.55,
+            'FC': 0.729
+        },
+        'WAT': {'HY': 0.428},
+        'BIT': {
+            'ST': 0.536,
+            'CT': 0.536,
+            'GT': 0.536,
+            'IC': 0.536
+        },
+        'WDS': {'ST': 0.493},
+        'RFO': {
+            'IC': 0.019,
+            'ST': 0.142,
+            'CT': 0.019
+        },
+        'SUN': {'PV': 26.1},
+        'KER': {'GT': 1.0},
+        'PC': {'ST': 0.142},
+        'PG': {'ST': 1.0},
+        'SUB': {'ST': 0.436},
+        'LFG': {
+            'CA': 0.733,
+            'CT': 0.733,
+            'IC': 0.733,
+            'GT': 0.733
+        },
+        'MWH': {'BA': 1.0},
+        'OBS': {'ST': 0.493},
+        'WND': {'WT': 0.374},
+        'OBL': {'IC': 0.493}
+    }
+
+    return cf
+
+
+def university_capacity():
+    """
+
+    This function uses previous functions to generate
+    a dictionary containing an estimate for the total
+    capacity of each university in 2018.
+
+    Returns:
+    --------
+    capacity: dict
+        A dictionary with each university as a key and
+        an estimate of the total capacity at the value in
+        MW.
+
+    """
+
+    data = generation('2018')
```

Comment:
```suggestion
    data = generation(year=year, path=path, prefix=prefix)
```

Does this function need to be for 2018 only? It might be interesting to see how the capacity evolves over time.

Reply:
I did it for 2018 because I could not find accurate data for capacity factors for the earlier years (and I know they are evolving by year).

Reply:
Got it, we can still set the default value to "2018" rather than hard coding it.

Reply:
In my `plot_data` function for the capacity argument, I have it raise an error if 2018 is not selected as the year. Should I remove this too in case someone wanted to make approximations based on current capacity factors?

Reply:
Yeah, that'd be good. I couldn't catch every instance, but where you can make a function generic,  you should.